packages/next/src/server/app-render/create-component-tree.tsx

```diff
@@ -97,6 +97,11 @@ async function createComponentTreeInternal({
     query,
   } = ctx
 
+  const { url } = staticGenerationStore
+  if (!url) {
```

Comment:
We don't need these `!url` error as `url` is already typed as string. Same as the Invalid URL error. I'd try to avoid adding new errors might shown to users that are not able to test but might break when some where cases pop up.

This have happened before in streaming rendering, where users hit the weird condition and trigger the error. As it's not for facing users, more like we could have a bad case of url and lead to this error. It's better to not introduce new errors while it's just refactoring.

Reply:
The `url` type here in the server is not typed as a string though. The case where this would be hit is if this code path is ran during `generateStaticParams` which does not have a `url` yet, only the `page`.  I'll investigate making the types stricter for this case.